src/dymaptic.GeoBlazor.Core/Components/LocationService.cs

```diff
@@ -152,133 +157,1101 @@ public async Task<List<AddressCandidate>> AddressesToLocations(List<Address> add
+    /// <param name="outFields">
+    ///     The list of fields included in the returned result set. This list is a comma delimited list of field names. If you specify the shape field in the list of return fields, it is ignored. For non-intersection addresses you can specify the candidate fields as defined in the geocode service. For intersection addresses you can specify the intersection candidate fields.
+    /// </param>
     /// <param name="outSpatialReference">
     ///     The spatial reference of the output geometries. If not specified, the output geometries are in the spatial reference of the input geometries when performing a reverse geocode and in the default spatial reference returned by the service if finding locations by address.
     /// </param>
+    /// <param name="searchExtent">
+    ///     Defines the extent within which the geocode server will search. Requires ArcGIS Server version 10.1 or greater.
+    /// </param>
     /// <param name="requestOptions">
-    ///     Additional options to be used for the data request
+    ///     Additional options to be used for the data request 
     /// </param>
-    public async Task<List<AddressCandidate>> AddressesToLocations(string url, List<Address> addresses,
-        string? countryCode = null, List<string>? categories = null, LocationType? locationType = null,
-        SpatialReference? outSpatialReference = null, RequestOptions? requestOptions = null)
+    /// 
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
+    public async Task<List<AddressCandidate>> AddressToLocations(string url, Address address,
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
+        List<string>? categories = null, string? countryCode = null, bool? forStorage = null, Point? location = null,
+        LocationType? locationType = null, string? magicKey = null, int? maxLocations = null,
+        List<string>? outFields = null, SpatialReference? outSpatialReference = null, Extent? searchExtent = null,
+        RequestOptions? requestOptions = null)
     {
-        IJSStreamReference streamRef = await InvokeAsync<IJSStreamReference>("addressesToLocations", url, addresses, countryCode, categories, locationType, outSpatialReference, requestOptions);
-        await using Stream stream = await streamRef.OpenReadStreamAsync(1_000_000_000L);
-        using var ms = new MemoryStream();
-        await stream.CopyToAsync(ms);
-        ms.Seek(0, SeekOrigin.Begin);
-        string json = Encoding.UTF8.GetString(ms.ToArray());
+        IJSStreamReference streamRef = await InvokeAsync<IJSStreamReference>("addressToLocations", url, address,
+            categories, countryCode, forStorage, location, locationType, magicKey, maxLocations, outFields,
+            outSpatialReference, searchExtent, requestOptions);
+        var json = await ConverJsStreamToString(streamRef);
```

Comment:
Same issue here, we broke this one by inserting an expected parameter in the middle of the JS parameters.